pkg/scheduler/scheduling_queue_test.go

```diff
@@ -688,11 +694,296 @@ func TestHeadroom(t *testing.T) {
+
+	headRoom = leaf1.getMaxHeadRoom()
+	assert.Assert(t, headRoom == nil, "headRoom of leaf1 should be nil because no max set for all queues")
+
+	headRoom = leaf2.getMaxHeadRoom()
+	assert.Assert(t, headRoom == nil, "headRoom of leaf2 should be nil because no max set for all queues")
+
+	// recreate the structure, set max capacity only in parent level
+	// structure is:
+	// root (max: nil)
+	//   - parent (max: 20,8)
+	//     - leaf1 (max: nil)  (alloc: 5,3)
+	//     - leaf2 (max: nil)  (alloc: 6,4)
+	root, err = createRootQueue(nil)
+	resMap := map[string]string{"first": "20", "second": "8"}
+	assert.NilError(t, err, "failed to create root queue with limit")
+	parent, err = createManagedQueue(root, "parent", true, resMap)
+	assert.NilError(t, err, "failed to create parent queue")
+	leaf1, err = createManagedQueue(parent, "leaf1", false, nil)
+	assert.NilError(t, err, "failed to create leaf1 queue")
+	leaf2, err = createManagedQueue(parent, "leaf2", false, nil)
+	assert.NilError(t, err, "failed to create leaf2 queue")
+
+	// allocating and allocated
+	res, err = resources.NewResourceFromConf(map[string]string{"first": "1", "second": "1"})
+	assert.NilError(t, err, "failed to create resource")
+	leaf1.incAllocatingResource(res)
+	leaf2.incAllocatingResource(res)
+	leaf2.incAllocatingResource(res)
+	res, err = resources.NewResourceFromConf(map[string]string{"first": "4", "second": "2"})
+	assert.NilError(t, err, "failed to create resource")
+	err = leaf1.QueueInfo.IncAllocatedResource(res, true)
+	assert.NilError(t, err, "failed to set allocated resource on leaf1")
+	err = leaf2.QueueInfo.IncAllocatedResource(res, true)
+	assert.NilError(t, err, "failed to set allocated resource on leaf2")
+
+	// root headroom should be nil
+	headRoom = root.getMaxHeadRoom()
+	assert.Assert(t, headRoom == nil, "headRoom of root should be nil because no max set for all queues")
+
+	// parent headroom = parentMax - leaf1Allocated - leaf2Allocated
+	// parent headroom = (20 - 5 - 6, 8 - 3 - 4) = (9, 1)
+	res, err = resources.NewResourceFromConf(map[string]string{"first": "9", "second": "1"})
+	headRoom = parent.getMaxHeadRoom()
+	if err != nil || !resources.Equals(res, headRoom) {
+		t.Errorf("parent queue head room not as expected %v, got: %v (err %v)", res, headRoom, err)
+	}
+
+	// leaf1 headroom = parentMax - leaf1Allocated - leaf2Allocated
+	headRoom = leaf1.getMaxHeadRoom()
+	if !resources.Equals(res, headRoom) {
+		t.Errorf("leaf1 queue head room not as expected %v, got: %v (err %v)", res, headRoom, err)
+	}
+
+	headRoom = leaf2.getMaxHeadRoom()
+	if !resources.Equals(res, headRoom) {
+		t.Errorf("leaf2 queue head room not as expected %v, got: %v (err %v)", res, headRoom, err)
+	}
+
+	// recreate the structure, set max capacity in root level, verify that will be ignored
+	// structure is:
+	// root (max: 1, 1)
+	//   - parent (max: 20,8)
+	//     - leaf1 (max: nil)  (alloc: 5,3)
+	//     - leaf2 (max: nil)  (alloc: 6,4)
```

Comment:
This does not add anything that is/cannot tested by calling getMaxHeadroom on the root queue directly.
Should be rolled into the `nil` tests at the start.
Please remove this case.

Reply:
removed